lib/src/component_declaration/component_base.dart

```diff
@@ -216,9 +216,11 @@ abstract class UiComponent<TProps extends UiProps> extends react.Component with
 
   /// Validates that props with the `@requiredProp` annotation are present.
   void validateRequiredProps(Map appliedProps) {
+    final requiredPropNamesToSkipValidation = newProps().requiredPropNamesToSkipValidation;
     consumedProps?.forEach((consumedProps) {
       consumedProps.props.forEach((prop) {
         if (!prop.isRequired) return;
+        if (requiredPropNamesToSkipValidation.contains(prop.key.split('.').last)) return;
```

Comment:
Hmm I'm not a big fan of pulling the name out of the prop key here is a good idea, since it won't behave well for edge-cases:
- props with the same name declared in different prop mixins
- props with custom keys that don't match the name

Some other paths forward that I can think of off the top of my head, in order from most to least preference:
- Update PropDescriptor to be able to tell the difference between `@requiredProp` and `late`, and just disable this validation in the component for `late` props entirely (since they're already validated by `UiProps.call`/`.build`)
   -  Maybe this is a new field like `PropDescriptor.isLate`?
- Update PropDescriptor to include the prop name and use that in check
- Remove this old automatic required prop validation (in favor of the new validation in `UiProps.call`/`.build` and the analyzer plugin)

Thoughts?

Reply:
Yeahh 😕. I'm not 100% sure how I feel about that option, but my thought was if, as part of consumers migrating to null safety, they'll be getting rid of `@reuiredProp`, then maybe it's worth it to safe us some headaches in getting this working.

But, thinking more about that... we probably have consumers relying on that behavior now in tests... My thought was that if they were only using it as a debugging tool during development, we could probably remove it without breaking anyone, but if it'll cause existing tests to fail, then it's probably not worth the added friction after all.

Reply:
I just updated it to add an `isLate` property to `PropDescriptor`, but I realized that turning off proptype validation for all late props means that late props in old boilerplates won't have any validation since the validation in call/build only runs in the newest boilerplates, right? Are we cool with that?

Reply:
Yeah I'm cool with that 👍